native/react-native/common-components/modals/confirm-modal.js

```diff
@@ -0,0 +1,185 @@
+import React, { Component } from "react";
+import { Modal, Text, TouchableHighlight, View, StyleSheet } from "react-native";
+import { connect } from "react-redux";
+import { PropTypes } from "prop-types";
+import { configureConfirmModal } from "../../redux"; //This action creator is included in this folder ./modalActions.js
+
+const mapStateToProps = state => ({
+    modalState: state.modalState
+});
+
+/**
+ * Basic shape of state.modalState (referenced above in mapStateToProps), which should go in your reducer for this Redux property:
+ *    confirmModalState: {
+        isConfirmModalVisible: false,
+        titleMessage: "",
+        bodyMessage: "",
+        confirmFunction: null,
+        confirmMessage: "",
+        cancelMessage: ""
+    };
+ */
+
+const mapDispatchToProps = dispatch => ({
+    configureConfirmModal: (modalStateObject) => dispatch(configureConfirmModal(modalStateObject))
+});
+
+//We use this to reset the modal when we're done with it.
+const defaultModalState = {
```

Comment:
I used to have an action to reset the modal state, but doing it here saves redux code. Do you have a preference?

Reply:
I've done what @Fleury14 has as well. Specially since somewhere else in the app (default Redux state?), this state has to be originally set.

Reply:
Yeah, above this there is an example thats commented out of my redux resetter. Probably should have made that clearer though.